docs/arch/02-core-concepts.md

```diff
@@ -0,0 +1,693 @@
+- Network: `pkg/permissions/profile.go`
+- Mount declarations: `pkg/permissions/profile.go`
+
+**Related concepts:** RunConfig, Workload, Security
+
+### Group
+
+A **group** is a logical collection of MCP servers that share a common purpose or use case.
+
+**Use cases:**
+- Organizational structure (e.g., "data-analysis" group)
+- Virtual MCP servers (aggregate multiple MCPs into one)
+- Access control (apply policies at group level)
+- Client configuration (configure clients to use groups)
+
+**Operations:**
+- Create group: `thv group create <name>` or add workloads with `--group` flag
+- List all groups: `thv group list`
+- List workloads in group: `thv list --group <name>`
+- Remove group: `thv group rm <name>`
+
+**Implementation:**
+- Group management: `pkg/groups/`
+- Workload group field: `pkg/runner/config.go`
+
+**Related concepts:** Virtual MCP, Workload, Client
+
+### Registry
+
+A **registry** is a catalog of MCP server definitions with metadata, configuration, and provenance information.
+
+**Registry types:**
+
+1. **Built-in Registry**: Curated by Stacklok
+   - Source: https://github.com/stacklok/toolhive-registry
+   - Embedded in the binary
+   - Trusted and verified servers
+
+2. **Custom Registry**: User-provided
+   - Configured via config file
+   - JSON file or remote URL
+   - Organization-specific servers
+
+3. **Registry API**: Self-hosted registry server
+   - Binary: `thv-registry-api`
+   - Storage: File or Kubernetes ConfigMap
+   - API endpoints for management
+
+**Registry entry types:**
+- `servers` - Container-based MCP servers
+- `remoteServers` - Remote MCP servers (HTTPS endpoints)
+- `groups` - Predefined groups of servers
+
+**Implementation:**
+- Registry types: `pkg/registry/types.go`
+- Registry manager: `pkg/registry/registry.go`
+- API server: `cmd/thv-registry-api/`
+
+**Related concepts:** Image Metadata, Remote Server Metadata
+
+### Session
+
+A **session** tracks state for MCP client connections, particularly for transports that require session management.
+
+**Session types:**
+
+1. **SSE Session**: For stdio transport with SSE proxy mode
+   - Tracks connected SSE clients (multiple clients can connect, but share single stdio connection to container)
+   - Message queue per client
+   - Endpoint URL generation
+   - Note: stdio transport has single connection/session to container
+
+2. **Streamable Session**: For stdio transport with streamable proxy mode
+   - Tracks `Mcp-Session-Id` header
+   - Request/response correlation
+   - Ephemeral sessions for sessionless requests
+
+3. **Proxy Session**: For transparent proxy (SSE/Streamable transports)
+   - Session ID detection from headers
+   - Session ID detection from SSE body
+   - Minimal state tracking
+
+**Session lifecycle:**
+- Created on first request or explicit initialize
+- Tracked via session manager with TTL
+- Cleaned up after inactivity or explicit deletion
+
+**Implementation:**
+- Manager: `pkg/transport/session/manager.go`
```

Comment:
File path reference is incorrect: `pkg/transport/session/manager.go` does not exist.

Session management is embedded in individual session implementations:
- `pkg/transport/session/sse_session.go`
- `pkg/transport/session/streamable_session.go`
- `pkg/transport/session/proxy_session.go`

Consider removing this reference or noting that session management is distributed across the individual session files.

Evidence: Verified via codebase exploration